src/compiler/axis.js

```diff
@@ -101,9 +101,9 @@ axis.grid = function(def, name, encoding, layout) {
           value: def.offset
         },
         x2: {
+          field: {group: 'mark.group.width'},
```

Comment:
Is this referencing a grandparent's width? I'd recommend `field: {group: 'width', level: 2}` in this case, rather than manually walking up the scenegraph structure. See [value refs](https://github.com/vega/vega/wiki/Marks#value-references) documentation for more detail.